magento-research/pwa-studio
# [feature]: Toasts

# TODO LIST

- [x] ~CSS Transitions for toast addition and removal~ Decided that pop-in/out was fine.
- [x] Refactor error notification work from #871 to use toasts.
- [x] If identical toast is repeated _reset timeout_ and _flash the opacity_ to indicate reset
- [x] Dismissable toasts should also auto remove after _7 seconds_
- [x] Show online/offline indicator _along with_ online/offline toast.
- [x] Add a warning type (yellow?) toast.
- [x] export from peregrine properly
- [x] Cleanup (console logs, etc)
- [x] Handle desktop toasts (currently always centered) cc @soumya-ashok 

## Description

This PR brings you 🥁 🥁 🥁 🥁 🥁 🥁 toasts 🍞 !

Big changes:

1) Addition of a toast context/store along with hooks that enable usage of the store
1) Implementation of a "toast container" and toasts in Venia (html/css/etc)
1) Refactoring of error notifications to use toasts.
1) Refactoring of online indicator to use toasts.
1) Refactoring of online indicator to live in the header as just an offline icon (when offline).

Toasts should:

* Look delicious.
* Be removable by `timeout`, or some default (5 seconds).
* Have different states (info, error, etc)
* Be tested!

[![Image from Gyazo](https://i.gyazo.com/2a6b7fc3fe20158690a8fa843c87574d.gif)](https://gyazo.com/2a6b7fc3fe20158690a8fa843c87574d)

[![Image from Gyazo](https://i.gyazo.com/c8c60b14b454ac22abd77ba373fa4c50.gif)](https://gyazo.com/c8c60b14b454ac22abd77ba373fa4c50)

## Related Issue





Closes #783.

## Verification Steps

1. Open the app. Enable offline mode in dev tools. See offline toast and icon in header!
1. Disable offline mode. See online toast!
1. Add an error to `miniCart.js` and rerender. See an error toast!

## How Have YOU Tested this?
Manually, and with verification steps, and wrote unit tests.


## Proposed Labels for Change Type/Package




## Checklist:


- [ ] I have updated the documentation accordingly, if necessary. paging @jcalcaben! Should we make changes to docs for toasts?
